Add updated documentation for overflow in filter patterns (#2340)
hashicorp/design-system: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md

This can be done by adding a "Clear all" action to the [Application State](/components/application-state) or by including dismissible [Tags](/components/tag) for each applied filter.

## Filter overflow

Depending on the data complexity and expected user interactions, it may be necessary to consider how filter pattern elements might overflow into other areas of the UI.

- What happens when there are many filters applied to a data set?
- Is there a specific hierarchy or importance in the filter parameters that might determine how they are ordered or displayed to the user?
- What happens when there are many parameters that can be filtered upon?

> These scenarios can clutter the UI, detracting from a table's primary purpose: organizing data categorically, relationally, and structurally.

Reviewer: 💯

2. Moving less important filter parameters to a [Flyout](/components/flyout), using [Checkbox groups](/components/form/checkbox) for each parameter and its values.

This approach offers a scalable solution for highly complex data sets while communicating a natural hierarchy of filter importance.

![Trigger the overflow of filter parameters with a Button](/assets/patterns/filter-patterns/filter-parameters-overflow-more-filters.png)

<Doc::ImageCaption @text="Use a secondary Button to trigger a Flyout." />

![Display the overflow of filter parameters within a Flyout](/assets/patterns/filter-patterns/filter-parameters-overflow-flyout.png)

<Doc::ImageCaption @text="Display the overflow of filter parameters in a Flyout" />

#### Accordions within a Flyout

For excessive filter parameters causing a long scroll in the [Flyout](/components/flyout), consider placing each parameter in a `flush`, `small` [Accordion](/components/accordion), and optionally, add an "Expand/Collapse All" Button. This approach gives users more control over the UI and allows for visual comparison between filter parameters.

> ![Accordions within a Flyout containing filter parameters](/assets/patterns/filter-patterns/filter-parameters-overflow-flyout-accordion.png)

Reviewer 2: [nit non-blocking] Would it be helpful to show the expand/collapse all in this image?

Author: Good idea, I've added this!

<Doc::ImageCaption @text="Use Accordions within a Flyout to support an excessive number of filter parameters." />

### Filter hierarchy

> Filter parameters vary in importance, which should be reflected in their organization in the UI. Moving frequently used filters to a [Flyout](/components/flyout) can be detrimental, as accessing common filters may require multiple steps.

Reviewer 3: [nit non-blocking]
```suggestion
Filter parameters vary in importance, which should be reflected in their position within the UI. Moving frequently used filters to a [Flyout](/components/flyout) can be detrimental, as accessing them may require multiple steps.
```

!!! Info

Impact and priority of filters should be determined by each product and application team. User needs and priorities vary across products and user journeys, affecting the prioritization of UI elements.

> !!!

Reviewer 3: [suggestion non-blocking] This feels obvious because all of this is purely recommendation-based and needs to be built by the product teams. I wonder if it needs to be said at all, but I'm curious what others think.

Author: Yeah, there's a bit of implication here, I'll remove it.

![Trigger the overflow of filter parameters with a Button](/assets/patterns/filter-patterns/filter-parameters-overflow-more-filters.png)

<Doc::ImageCaption @text="Use a secondary Button to trigger a Flyout." />

![Display the overflow of filter parameters within a Flyout](/assets/patterns/filter-patterns/filter-parameters-overflow-flyout.png)

<Doc::ImageCaption @text="Display the overflow of filter parameters in a Flyout" />

#### Accordions within a Flyout

For excessive filter parameters causing a long scroll in the [Flyout](/components/flyout), consider placing each parameter in a `flush`, `small` [Accordion](/components/accordion), and optionally, add an "Expand/Collapse All" Button. This approach gives users more control over the UI and allows for visual comparison between filter parameters.

![Accordions within a Flyout containing filter parameters](/assets/patterns/filter-patterns/filter-parameters-overflow-flyout-accordion.png)

<Doc::ImageCaption @text="Use Accordions within a Flyout to support an excessive number of filter parameters." />

> ### Filter hierarchy

Reviewer 3: [suggestion non-blocking] This section feels like it re-iterates lines 212-213. Could it be combined with that content somehow?

Author: I think I'll just kill this whole section since it's referenced in other places

Filter parameters vary in importance, which should be reflected in their organization in the UI. Moving frequently used filters to a [Flyout](/components/flyout) can be detrimental, as accessing common filters may require multiple steps.

!!! Info

Impact and priority of filters should be determined by each product and application team. User needs and priorities vary across products and user journeys, affecting the prioritization of UI elements.
!!!

### Putting it all together

> An effective filter pattern support overflow in common scenarios will consist of:

Reviewer 3: [issue non-blocking] This sentence is hard to follow.
What are you trying to communicate here?

Author: Updated this a bit to be more clear
